cirq-core/cirq/vis/heatmap_test.py

```diff
@@ -343,3 +344,171 @@ def test_plot_updates_local_config():
         _, ax = plt.subplots()
         random_heatmap.plot(ax)
         assert ax.get_title() == original_title
+
+
+@pytest.mark.usefixtures('closefigures')
+def test_heatmap_plot_highlighted_qubits():
+    value_map = {
+        (grid_qubit.GridQubit(0, 0),): 0.1,
+        (grid_qubit.GridQubit(0, 1),): 0.2,
+        (grid_qubit.GridQubit(0, 2),): 0.3,
+        (grid_qubit.GridQubit(1, 0),): 0.4,
+    }
+    single_qubit_heatmap = heatmap.Heatmap(value_map)
+
+    qubits = np.array([k[0] for k, _ in value_map.items()])
+    highlighted_qubits_indices = [1, 3]
+    highlighted_qubits = qubits[highlighted_qubits_indices]
+
+    expected_linewidths = [2, 4, 2, 4]
+    expected_edgecolors = np.vstack(
+        (
+            to_rgba_array('grey'),  # grey
+            to_rgba_array('red'),  # red
+            to_rgba_array('grey'),  # grey
+            to_rgba_array('red'),  # red
+        )
+    )
+    # list of tuples: (offset, onoffseq), onoffseq = None for solid line.
+    expected_linestyles = [(0.0, [7.4, 3.2]), (0, None), (0.0, [7.4, 3.2]), (0, None)]
```

Comment:
Can you replace the `[7.4, 3.2]` value with appropriate matplotlib call?
We don't want this to fail if matplotlib changes their dashed styling.

Reply:
Hmm, I'm unsure if this can be done by a simple matplotlib call since these on-off sequence values controlling the linestyle also depend on the linewidth set by the user. A simple way could be to create a dummy heatmap with properties set the same as the properties when qubit highlighting is enabled, and then compare them. What do you think?

Reply:
Sounds good.  The linestyles however seem to be resolved in [matplotlib.Collection](https://github.com/matplotlib/matplotlib/blob/14a04a02789e5deab2325531b46d570f16a29986/lib/matplotlib/collections.py#L618) so a local helper function in heatmap_test.py should do:

```py
def _to_linestyle_tuple(linestyles, linewidths=None):
    c = mpl.collections.Collection(linestyles=linestyles, linewidths=linewidths)
    return c.get_linestyles()[0]
```

Reply:
Great, thanks! Have made the change.